pkg/logs/tail.go

```diff
@@ -0,0 +1,603 @@
+			}
+			io.Copy(ioutil.Discard, resp.Body)
+			// Try the response again after a 1 second wait.
+			if resp.StatusCode/100 == 5 && resp.StatusCode != 501 ||
+				resp.StatusCode == 429 {
+				time.Sleep(1 * time.Second)
+				continue
+			}
+			// Failing at this point is unrecoverable,
+			return
+		}
+
+		// Read and parse response, send batches to the output loop.
+		scanner := bufio.NewScanner(resp.Body)
+		// Use a 10MB buffer for the bufio scanner, as we don't know
+		// how big some of the responses will be.
+		const tmb = 10 << 20
+		buf := make([]byte, tmb)
+		scanner.Buffer(buf, tmb)
+
+		for scanner.Scan() {
+			// Scan one line at a time, and get only one batch
+			// at a time.
+			b := scanner.Bytes()
+			batch, err := parseResponseData(b)
+			if err != nil {
+				// We can't parse the response, attempt to
+				// re-request from the last window & batch.
+				text.Warning(out, "unable to parse resp body: %v", err)
+				path = makeNewPath(out, path, curWindow, lastBatchID)
+				continue
+			}
+
+			// If we got a batch back, there will be an ID.
+			if batch.ID != "" {
+				// Record last batchID in case
+				// anything fails along the way, we
+				// can re-request.
+				lastBatchID = batch.ID
+				// Send batch down batchCh to the output loop.
+				c.batchCh <- batch
+			}
+
+		}
+		resp.Body.Close()
+
+		if err := scanner.Err(); err != nil {
+			// ErrUnexpectedEOFs need to be retried, but they
+			// produce a lot of noise for the user, so don't log.
+			if err != io.ErrUnexpectedEOF {
+				text.Warning(out, "error scanning response body: %v", err)
+			}
+			// Something happened in the scanner, re-request the
+			// current batchID.
+			path = makeNewPath(out, path, curWindow, lastBatchID)
+			continue
+		}
+
+		// Get our next time window to request.
+		_, next := getLinks(resp.Header)
+		curWindow, err = getTimeFromLink(next)
+		if err != nil {
+			text.Error(out, "error generating window from next link")
+		}
+
+		// We do NOT want to specify a batchID, as this
+		// request was successful.
+		lastBatchID = ""
+		path = makeNewPath(out, path, curWindow, lastBatchID)
+	}
+
+}
+
+// adjustTimes adjusts the passed in from and to flags based on the
+// specified padding.
+func (c *TailCommand) adjustTimes() {
+	if c.cfg.from != 0 {
+		// Adjust from based on search padding, we want to
+		// look back further.
+		c.cfg.from = c.cfg.from - int64(c.cfg.searchPadding.Seconds())
+	}
+
+	if c.cfg.to != 0 {
+		// Adjust to based on search padding, we want look forward more.
+		c.cfg.to = c.cfg.to + int64(c.cfg.searchPadding.Seconds())
+	}
+}
+
+// enableManagedLogging enables managed logging in our API.
+func (c *TailCommand) enableManagedLogging(out io.Writer) error {
+	_, err := c.Globals.Client.CreateManagedLogging(&c.Input)
+	if err != nil && err != fastly.ErrManagedLoggingEnabled {
+		return err
+	}
+
+	text.Success(out, "Managed logging enabled on service %s", c.Input.ServiceID)
```

Comment:
Do we also want to remove this message (for the same reason as removing the other message), and if so this is something else that should be removed/reflected in the blog post.

Reply:
I don't think we should.  This will go out and make the call every time, sometimes with a large delay (I've noticed).  Without the message, it makes for an odd silence in the CLI output.  We can make a -q (--quiet) flag down the road if people want a perfectly clean output they can pipe to something else.